### ISSUE TITLE
SingleLiveData doesn't work as expected

### ISSUE THREAD

**Issue by Ghedeon** [Shopify/livedata-ktx#19]
Failing test:
```kotlin
    @Test
    fun `single reattach`() {
        val liveData = SingleLiveData<Int>()
        val actual = mutableListOf<Int?>()
        val observer: (t: Int?) -> Unit = { actual.add(it) }
        
        liveData.observe(this, observer)
        liveData.value = 1
        liveData.removeObserver(observer)
        actual.clear()
        liveData.observe(this, observer)
        liveData.value = 2
        
        val expected = listOf(2)
        assertEquals(expected, actual) // ERROR: actual is [2, 2]
    }
```
The same test is passed by the original `SingleLiveEvent`: https://github.com/googlesamples/android-architecture/blob/dev-todo-mvvm-live/todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/SingleLiveEvent.java

Common use-case that leads to this: 
1) The view is subscribed to `SingleLiveData` and gets an event
2) The view is resubscribed (navigate to a different screen and come back)
3) The next event is received twice.

**Comment by Ghedeon** [comment]
Ehm... but it still fails.

**Comment by henrytao-me** [comment]
Hi @Ghedeon, sorry for late response. The problem has been fixed in release `2.0.2`.
